docs/tutorials/09-complete-tutorials/04-zkverify-js.md

```diff
@@ -0,0 +1,240 @@
+```
+</TabItem>
+<TabItem value="noir" label="Noir">
+```js
+const fs = require("fs");
+const proof = fs.readFileSync("../hello_world/proof.hex").toString()
+const public = fs.readFileSync("../hello_world/pub.hex").toString()
+```
+</TabItem>
+</Tabs>
+
+Once you have all the requirements imported, we will start by instantiating a session with our Volta testnet with an account(This account should have $tVFY to pay for transactions). 
+```js
+const session = await zkVerifySession.start().Volta().withAccount("seed-phrase")
+```
+
+For Circom and Noir proofs, we need to register a verification key on our Volta testnet which will be used in further steps while verifying proofs. This step is not required for Risc Zero because we already got a hash of the image id which can be used directly. You can execute the following code snippet to register a vkey:
+
+<Tabs groupId="register-vkey">
+<TabItem value="circom" label="Circom">
+```js
+const {events, regResult} = await session.registerVerificationKey().groth16({library: Library.snarkjs, curve: CurveType.bn128}).execute(key);
+
+events.on(ZkVerifyEvents.Finalized, (eventData) => {
+    console.log('Registration finalized:', eventData);
+    fs.writeFileSync("vkey.json", JSON.stringify({"hash": eventData.statementHash}, null, 2));
+    return eventData.statementHash
+});
+```
+
+Once registered you can find a new file called ``vkey.json`` with your registered verification key in the following format
+```json
+{
+  "vkey": "0x828c736b33ab492251a8b275468a29ce06e98fc833c0c7f0bc7f6272b300c05b"
+}
+```
+
+</TabItem>
+<TabItem value="noir" label="Noir">
+```js
+let vk = fs.readFileSync("../hello_world/vk.hex").toString()
+
+const {events, regResult} = await session.registerVerificationKey().ultraplonk().execute(vk.split("\n")[0]);
+
+events.on(ZkVerifyEvents.Finalized, (eventData) => {
+    console.log('Verification finalized:', eventData);
+    fs.writeFileSync("vkey.json", JSON.stringify({"hash": eventData.statementHash}, null, 2));
+    return eventData.statementHash
+});
+
+```
+Once registered you can find a new file called ``vkey.json`` with your registered verification key in the following format
+```json
+{
+  "vkey": "0x828c736b33ab492251a8b275468a29ce06e98fc833c0c7f0bc7f6272b300c05b"
+}
+```
+</TabItem>
+</Tabs>
+
+Next we will send a proof verification request to the Volta testnet, with all the details like which proving schema, proof, public signals and the key. We will also need to specify the ``domainId`` for which we want this proof to be aggregated. You can check more about Domain and Aggregation [here](../../overview/04-proof-aggregation/01-overview.md). For this tutorial, choose the Domain ID based on the target chain where you want to verify the attestations [List of existing domains](../../overview/04-proof-aggregation/05-domain-management.md). We will also create an event listener, to listen to the ``NewAggregationReceipt`` event whenever our proof is aggregated :- 
```

Comment:
Please note that domainID is optional. If not specified, proof will only be verified on zkVerify and nothing else